src/common/styles.c

```diff
@@ -580,6 +582,8 @@ void dt_styles_apply_to_selection(const char *name, gboolean duplicate)
   while(sqlite3_step(stmt) == SQLITE_ROW)
   {
     const int imgid = sqlite3_column_int(stmt, 0);
+    if(mode == 1)
```

Comment:
Can you please add an enum in styles.h with DT_STYLE_HISTORY_APPEND and DT_STYLE_HISTORY_OVERWRITE  and use DT_STYLE_HISTORY_OVERWRITE instead of 1 here? It is always better to use named number as it makes reading the code clear.